platform_channels/android/app/src/main/kotlin/dev/flutter/platform_channels/MainActivity.kt

```diff
@@ -50,8 +49,31 @@ class MainActivity : FlutterActivity() {
                 .setMessageHandler { message, reply ->
                     if (message == "getImage") {
                         val inputStream: InputStream = assets.open("eat_new_orleans.jpg")
-                        reply.reply(inputStream.readBytes());
+                        reply.reply(inputStream.readBytes())
                     }
                 }
+
+        val petList = mutableListOf<Map<String, String>>()
+        val gson = Gson()
+
+        // A BasicMessageChannel for sending petList to Dart.
+        val stringCodecChannel = BasicMessageChannel(flutterEngine.dartExecutor, "stringCodecDemo", StringCodec.INSTANCE)
+
+        // Registers a MessageHandler for BasicMessageChannel to receive pet details to be
+        // added in petList and send the it back to Dart using stringCodecChannel.
+        BasicMessageChannel(flutterEngine.dartExecutor, "jsonMessageCodecDemo", JSONMessageCodec.INSTANCE)
+                .setMessageHandler { message, reply ->
+                    petList.add(0, gson.fromJson(message.toString(), object : TypeToken<Map<String, String>>() {}.type))
+                    stringCodecChannel.send(gson.toJson(mapOf("petList" to petList)))
+                }
+
+        // Registers a MessageHandler for BasicMessageChannel to receive the index of pet
+        // details to be removed from the petList and send the petList back to Dart using
+        // stringCodecChannel.
+        BasicMessageChannel(flutterEngine.dartExecutor, "binaryCodecDemo", BinaryCodec.INSTANCE)
+                .setMessageHandler { message, reply ->
```

Comment:
I don't think it's possible given the UI you've built, but what happens if an invalid index is sent to this method? Does the error propagate correctly back to Dart?

Reply:
Yes, in the current UI it's not possible and there is no way that the error message is propagated correctly back to the Dart. I'll check whether the index exists or not and send a reply back to `BasicMessageChannel`. If reply is null that means the index doesn't exist and we can show the apt error message.